core/src/smartcontracts/wasm.rs

```diff
@@ -850,7 +824,24 @@ impl<'wrld> Runtime<state::SmartContract<'wrld>> {
         let mut store = self.create_store(state);
         let smart_contract = self.create_smart_contract(&mut store, bytes)?;
 
-        Self::execute_main_with_store(&smart_contract, &mut store)
+        let main_fn = Self::get_typed_func(
+            &smart_contract,
```

Comment:
What triggers me is that this is called a smart-contract and _not_ a trigger, while the entrypoint to what's being called by the trigger is called the _trigger main_.